08-CMSIS/Makefile

```diff
@@ -1,120 +1,75 @@
+	mkdir -p $($(1)_RELEASE_DIR)
+	$(CC) $(CFLAGS) $($(1)_CPU) -Wl,-T$($(1)_TARGET_ld) -nostartfiles \
+		  $(addprefix -I,$($(1)_INC_DIR)) \
+		  $(CORE_SOURCE) $($(1)_DEPENDENT_SOURCE) -o $($(1)_TARGET_elf)
+	$(CROSS_COMPILE)objcopy -Obinary $($(1)_TARGET_elf) $($(1)_TARGET_bin)
+	$(CROSS_COMPILE)objdump -S $($(1)_TARGET_elf) > $($(1)_TARGET_dump)
+endef
 
-STM32P103_TARGET_name           := $(STM32P103)_os
-STM32P103_TARGET                := $(STM32P103_TARGET_name).bin
-STM32P103_TARGET_ld             := $(STM32P103_PLATFORM_DIR)/os.ld
-STM32P103_TARGET_bin            := $(STM32P103_RELEASE_DIR)/$(STM32P103_TARGET_name).bin
-STM32P103_TARGET_elf            := $(STM32P103_RELEASE_DIR)/$(STM32P103_TARGET_name).elf
-STM32P103_TARGET_dump           := $(STM32P103_RELEASE_DIR)/$(STM32P103_TARGET_name).list
-###################
-
-
-
-# Target:STM32F429 device
-STM32F429                       := STM32F429
-STM32F429_RELEASE_DIR           := $(RELEASE_DIR)/$(STM32F429)
-STM32F429_PLATFORM_DIR          := $(PLATFORM_DIR)/$(STM32F429)
-STM32F429_CMSIS_DIR             := $(CMSIS_DIR)/$(STM32F429)
-
-STM32F429_DEPENDENT_HEADER_DIR  := $(STM32F429_PLATFORM_DIR)/$(INC_DIR)/
-STM32F429_DEPENDENT_SOURCE      := $(wildcard $(STM32F429_PLATFORM_DIR)/$(SRC_DIR)/*.c)
-STM32F429_INC_DIR               := $(CORE_HEADER_DIR) $(STM32F429_DEPENDENT_HEADER_DIR) $(CMSIS_DIR) $(STM32F429_CMSIS_DIR)
-
-STM32F429_TARGET_name           := $(STM32F429)_os
-STM32F429_TARGET                := $(STM32F429_TARGET_name).bin
-STM32F429_TARGET_ld             := $(STM32F429_PLATFORM_DIR)/os.ld
-STM32F429_TARGET_bin            := $(STM32F429_RELEASE_DIR)/$(STM32F429_TARGET_name).bin
-STM32F429_TARGET_elf            := $(STM32F429_RELEASE_DIR)/$(STM32F429_TARGET_name).elf
-STM32F429_TARGET_dump           := $(STM32F429_RELEASE_DIR)/$(STM32F429_TARGET_name).list
-###################
-
-
-all: $(STM32P103_TARGET) $(STM32F429_TARGET)
-
-
-$(STM32P103_TARGET): $(CORE_SOURCE) $(STM32P103_DEPENDENT_SOURCE)
-	mkdir -p $(STM32P103_RELEASE_DIR)
-	$(CC) $(CFLAGS) -mcpu=$(CORTEX_M3) -Wl,-T$(STM32P103_TARGET_ld) -nostartfiles \
-		  $(addprefix -I,$(STM32P103_INC_DIR)) \
-		  $^ -o $(STM32P103_TARGET_elf)
-	$(CROSS_COMPILE)objcopy -Obinary $(STM32P103_TARGET_elf) $(STM32P103_TARGET_bin)
-	$(CROSS_COMPILE)objdump -S $(STM32P103_TARGET_elf) > $(STM32P103_TARGET_dump)
-
-
-
-$(STM32F429_TARGET): $(CORE_SOURCE) $(STM32F429_DEPENDENT_SOURCE)
-	mkdir -p $(STM32F429_RELEASE_DIR)
-	$(CC) $(CFLAGS) -mcpu=$(CORTEX_M4) -Wl,-T$(STM32F429_TARGET_ld) -nostartfiles \
-		  $(addprefix -I,$(STM32F429_INC_DIR)) \
-		  $^ -o $(STM32F429_TARGET_elf)
-	$(CROSS_COMPILE)objcopy -Obinary $(STM32F429_TARGET_elf) $(STM32F429_TARGET_bin)
-	$(CROSS_COMPILE)objdump -S $(STM32F429_TARGET_elf) > $(STM32F429_TARGET_dump)
+#Create corresponding compiler arguments to the device
+$(eval $(call eval_compiler_command,$(STM32F429_DEVICE)))
+$(eval $(call eval_compiler_command,$(STM32P103_DEVICE)))
 
+.PHONY:erase cscope astyle clean
```

Comment:
The target "erase" is too STM32-specific! We should always think of the generic way.